src/main/java/io/rocketpartners/cloud/utils/SqlUtils.java

```diff
@@ -431,6 +431,60 @@ public static String buildInsertSQL(Connection conn, String tableName, Object[]
       return sql.toString();
    }
 
+   public static String buildInsertOnDuplicateKeySQL(Connection conn, String tableName, Object[] columnNameArray)
+   {
+      StringBuffer sql = new StringBuffer(buildInsertSQL(conn, tableName, columnNameArray));
+      sql.append(" ON DUPLICATE KEY UPDATE ");
+      for (Object col : columnNameArray)
+      {
+         sql.append(col).append("=VALUES(").append(col).append("), ");
```

Comment:
column names always need to be quoted with the db specific column quote character